src/github.com/turt2live/matrix-media-repo/controllers/upload_controller/upload_controller.go

```diff
@@ -68,8 +68,25 @@ func StoreDirect(contents io.Reader, contentType string, filename string, userId
 		return nil, err
 	}
 
+	allowed := false
 	for _, allowedType := range config.Get().Uploads.AllowedTypes {
-		if !glob.Glob(allowedType, fileMime) {
+		if glob.Glob(allowedType, fileMime) {
+			allowed = true
+		}
+	}
+	if !allowed {
+		exclusion := false
+		for user, userExcl := range config.Get().Uploads.AllowedExcl {
+			if user == userId {
```

Comment:
instead of iterating over the whole collection, we should only need to pull out the specific user's list and iterate over that.